src/shutdown.rs

```diff
@@ -225,7 +236,7 @@ impl SourceShutdownCoordinator {
         &mut self,
```

Comment:
The changed behavior of `let force_shutdown_tripwire = force_shutdown_tripwire.then(tripwire_handler);` changes the behavior of this returned `Future`, so that should be documented with something like: "Dropping this future prematurely will force shutdown the source."